File: atividades em sala/ts/first_class/bankAccount.ts


export class BankAccount {

    // attributes
    private id : string;
    private cpf : string;
    private balance : number;

    // constructors
    constructor(id: string, cpf : string, balance : number){
        this.id = id;
        this.cpf = cpf;
        this.balance = balance;
    }

    // getters and setters
    getId(){
        return this.id;
    }

    getCpf(){
        return this.cpf;
    }

    getBalance(){
        return this.balance;
    }

    // methods

    deposit(amount: number){
        this.balance += amount;
    }

    withdraw(amount : number){
        this.balance -= amount;
    }
}

function main(){

    // instance of bank account
    const account: BankAccount = new BankAccount("7624", "123.999.098-09", 2000.0);

    // show data
    showDataAccount(account, "\n> data before deposit:");

    // doing deposit
    account.deposit(260.0);

    // show data
    showDataAccount(account, "\n> data after deposit: ");

    // doing withdraw
    account.withdraw(100.0);

    // show data
    showDataAccount(account, "\n> data after withdraw: ");
}

function showDataAccount(account: BankAccount, message: string): void {
    console.log(message);
    console.log("\n| cpf: " + account.getCpf() + "\n| balance: " + account.getBalance().toFixed(2));
}

main();

File: atividades em sala/java/code/src/first_class/entities/BankAccount.java
package code.src.first_class.entities;

public class BankAccount {

    private int id;
    private String cpf;
    private Double balance;

    public BankAccount(){
    }

    public BankAccount(int id, String cpf, Double balance){
        this.id = id;
        this.cpf = cpf;
        this.balance = balance;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Double getBalance() {
        return balance;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public void deposit(Double amount){
        this.balance += amount;
    }

    public void withdraw(Double amount){
        this.balance -= amount;
    }

    public void showDetails(String message){
        System.out.println(message);
        System.out.println("\n| cfp: " + cpf + "\n| balance: R$ " + String.format("%.2f", balance));
    }
}
